data/core/common.lua

```diff
@@ -285,14 +414,27 @@ local function serialize(val, pretty, indent_str, escape, sort, limit, level)
   return tostring(val)
 end
 
--- Serialize `val` into a parsable string.
--- Available options
--- * pretty: enable pretty printing
--- * indent_str: indent to use ("  " by default)
--- * escape: use normal escape characters instead of the ones used by string.format("%q", ...)
--- * sort: sort the keys inside tables
--- * limit: limit how deep to serialize
--- * initial_indent: the initial indentation level
+
+---@class common.serializeoptions
+---@field pretty boolean Enables pretty printing.
+---@field indent_str string The indentation character to use. Defaults to `"  "`.
+---@field escape boolean Uses normal escape characters ("\n") instead of decimal escape sequences ("\10").
+---@field limit number Limits the depth when serializing nested tables. Defaults to `math.huge`.
+---@field sort boolean Sorts the output if it is a sortable table.
+---@field initial_indent number The initial indentation level. Defaults to 0.
+
+---Serialize a value into a Lua string that is loadable with load().
```

Comment:
```suggestion
---Serializes a value into a Lua string that is loadable with load().
```

We might want to add that this can break things like `math.huge` as it's transformed to `inf`.
Or maybe we should fix that in the function?